### ISSUE TITLE
Document collection concordance should have its own target

### ISSUE THREAD

**Issue by joeytakeda** [projectEndings/staticSearch#168]
As discussed in #157, the report no longer includes the concordance of all stems by default to mitigate memory issues. However, it makes sense to have the concordance to be a separate process that can be run at any point; most people won't want it with every build, but may want to run a concordance periodically (say during initial implementation of staticSearch or look at differences between versions, etc).

One approach:

* Create a new target in the build called concordance, which wouldn't be called from all or quick
* The concordance target would contain an XSLT that:
    * Gathers up all of the JSON stems
    * Parses them into XSLT maps (using `parse-json()`or some such)
    * And then uses those maps to determine frequency, averages, et cetera and outputs an HTML page (in the same directory as the report)

The assumption here is that the JSONs and the XSLT maps will be more efficiently stored in memory compared to the JSON; if that isn't true, then we'll have to see if there's a better approach (we could process each file one by one, outputting the info about each stem as a separate text file or something and then have a second XSLT that simply collects the text files, sorts them, and then puts them into the HTML page).

**Comment by martindholmes** [comment]
Adding a thought that @joeytakeda raised in email discussions earlier: we'll need to be sure that enough information is contained in the stem files, whatever the build settings, compared with using the tokenized HTML, which is what the current process uses.

**Comment by joeytakeda** [comment]
Starting a new branch for this: `iss-168-concordance`

**Comment by joeytakeda** [comment]
The JSON approach seems to work well—I'm able to build the a concordance in about a minute for the TEI set.

Wondering about this, though, and what's most useful. The HTML page for the TEI concordance is a hefty 12 MBs and much of it isn't particularly interesting. For me, at least, I only care about the top 10% (i.e. things that should possibly be stopwords) and the bottom 10% (things that might just be typos). And I think most people should look at the top 10% on one of their early runs to diagnose possible stopwords. 

What do you think about having this be four pages total: a "Most frequent words" page, a "least frequent words" page, a full wordlist page, and an index for them all?

**Comment by martindholmes** [comment]
I think if you're creating a full wordlist page anyway, there's no point in creating separate pages containing parts of that data; better just to have links and obvious visual cues giving easy access to the bits that are potentially most interesting.

**Comment by joeytakeda** [comment]
The main thing is that the page itself is huge and takes forever to navigate—at least in the case of TEI Guidelines concordance. But understandable that we don't want to create a bunch of additional stuff.

In the iss-168-concordance branch, I now have a working implementation of the concordance—once everything is built, you can do `ant -DssConfig=configTest.xml concordance`, which should build two things: an HTML concordance and a JSON version of it. I'm using a map as an intermediate stage for processing the table anyway, so I thought it made sense to provide the full JSON output since it may be useful for others. 

There's a bit of documentation to do there still before making a PR, but feel free to test it.

**Comment by martindholmes** [comment]
I've just tested and it works OK for the test, but when I try e.g.:

`ant -DssConfig=`pwd`/tei/config_tei.xml concordance`

or with the full path to the config file, or using the -DssConfigFile param, I just get the test data output, not a concordance for the tei stuff. What am I missing?

**Comment by joeytakeda** [comment]
You'll have to run the config task first to refresh config.xsl: 

```
ant -DssConfig=`pwd`/tei/config_tei.xml config concordance
```

(This is mostly a unique situation with the TEI build, since it's using the same set of XSLTs)

**Comment by martindholmes** [comment]
Agreed. A common use-case will be running a regular build, seeing an issue, and then running concordance to help with debugging, but other builds might have been run in the meantime and the original config file overwritten, so it might as well be generated again, since it's so quick.

**Comment by joeytakeda** [comment]
I've now added config as a dependency and added better documentation throughout (including documentation in the ODD--but did not regenerate the schema because I couldn't remember if that should be done in dev after a merge or in the branch itself).

I've also dropped the wordlist from the verbose report, since I think it's not necessary when we have the concordance. We could add a message that says "If you were expecting the wordlist, run ant concordance" when someone switches on the verbose report, if we think that's helpful. 

In any case, I've now made the PR since I believe all of the major work on this is completed.
